internal/saucecloud/cloud.go

```diff
@@ -376,3 +379,57 @@ func unregisterSignalCapture(c chan os.Signal) {
 	signal.Stop(c)
 	close(c)
 }
+
+// downloadArtifacts downloads artifact for job.
+func (r *CloudRunner) downloadArtifacts(artifactsCfg config.ArtifactDownload, jb job.Job) {
+	if !r.shouldDownloadArtifacts(artifactsCfg, jb) {
+		return
+	}
+
+	targetDir := filepath.Join(artifactsCfg.Directory, jb.ID)
+	if err := os.MkdirAll(targetDir, 0755); err != nil {
+		log.Error().Msgf("Unable to create %s to fetch artifacts (%v)", targetDir, err)
+		return
+	}
+
+	files, err := r.JobReader.GetJobAssetFileNames(context.Background(), jb.ID)
+	if err != nil {
+		log.Error().Msgf("Unable to fetch artifacts list (%v)", err)
+		return
+	}
+	for _, fileName := range files {
+		for _, pattern := range artifactsCfg.Match {
+			if glob.Glob(pattern, fileName) {
+				if err := r.doDownloadArtifact(targetDir, jb.ID, fileName); err != nil {
+					log.Error().Err(err).Msgf("Failed to download file: %s", fileName)
+				}
+				break
+			}
+		}
+	}
+}
+
+func (r *CloudRunner) doDownloadArtifact(targetDir, jobID, fileName string) error {
+	fileContent, err := r.JobReader.GetJobAssetFileContent(context.Background(), jobID, fileName)
+	if err != nil {
+		return err // TODO: Add ERROR log
```

Comment:
TODO? :D

Reply:
Forgot to remove that comment.
Logger action was done in the calling functions 👍